### DISCUSSION TITLE
Non-standard plugin issue fixers

### DISCUSSION THREAD

**Discussion by gready-hub** [vimeo/psalm#10881] [Plugin development]
Hey,

I've been creating a bunch of plugin's for our project and started playing around with the issue fixers but Psalm has a fixed set of supported issues that can't be modified or augmented. 

**vimeo/psalm/src/Psalm/Internal/Analyzer/ProjectAnalyzer.php**
```php
    public static function getSupportedIssuesToFix(): array
    {
        return array_map(
            /** @param class-string $issue_class */
            static function (string $issue_class): string {
                $parts = explode('\\', $issue_class);
                return end($parts);
            },
            self::SUPPORTED_ISSUES_TO_FIX,
        );
    }
```

My questions;
1. Is there a way to add supported issues fixers already that I missed?
2. Are the maintainers for/against the idea of augmenting the supported issues through the Plugin interface?
3. If you're for it, could you suggest a high level API that would be considered?
4. If you're not for it, quick reason why that is (could be misusing Psalm for all I know)

Thanks for your time peeps!

**Comment by orklah** [comment]
Hey!

An eternity ago, I made a few plugins and I was confronted with this issue among others.

I made a post about it here: https://github.com/vimeo/psalm/discussions/5261

At the time I wasn't familiar with Psalm so I didn't go forward on this but I still think there's lots of improvements to be made there

So yeah, if you want to go for it, go ahead!
